src/ser/mod.rs

```diff
@@ -74,31 +85,50 @@ pub fn to_string<S: Serialize>(value: &S) -> Result<String> {
     Ok(string)
 }
 
+/// A convenience method for serializing some object to a string.
+///
+/// With indent and add document declaration at the beginning of a document.
+pub fn to_pretty_string<S: Serialize>(value: &S) -> Result<String> {
+    // Create a buffer and serialize our nodes into it
+    let mut writer = Vec::with_capacity(128);
+    to_pretty_writer(&mut writer, value)?;
+
+    // We then check that the serialized string is the same as what we expect
+    let string = String::from_utf8(writer)?;
+    Ok(string)
+}
+
 /// An XML `Serializer`.
 pub struct Serializer<W>
 where
     W: Write,
 {
-    writer: W,
+    writer: EventWriter<W>,
 }
 
 impl<W> Serializer<W>
 where
     W: Write,
 {
-    pub fn new(writer: W) -> Self {
+    pub fn new(writer: EventWriter<W>) -> Self {
+        Self { writer: writer }
+    }
+
+    #[cfg(test)]
```

Comment:
Would it be better to rename this `new_without_xml_declaration()` or something similar? That way we can expose it to the user (if we want to) and it doesn't sound like we're making a special case simply for testing.